### DISCUSSION TITLE
Redirect/Rewrite on cookie _not_ being present

### DISCUSSION THREAD

**Discussion by kelko** [vercel/next.js#39990] [Ideas]
### Describe the feature you'd like to request

Currently one can redirect and rewrite based on headers, cookies, ... (https://nextjs.org/docs/api-reference/next.config.js/rewrites#header-cookie-and-query-matching and https://nextjs.org/docs/api-reference/next.config.js/redirects#header-cookie-and-query-matching) using `has`.

I would like the ability to check the _absence_ of a cookie to decide if a redirect (to one of those infamous disclaimer / GDPR pages) is necessary.

### Describe the solution you'd like

either: 

- one can set a `has_not` array, similar to `has`
- every  input has to match the `source` and all `has` and  also not any `has_not` so activate the redirect/rewrite

or: 

- one can pass a function as `has`.`value` to I can implement own match logic
- and as right now the input has to match match the `source` and all `has`, but now I can check for a has the absence of a cookie

### Describe alternatives you've considered

`<Script` with strategy `afterInteractive` which checks presence of the cookie and triggers the rewrite by code. Feels way more hacky.

**Comment by icyJoseph** [comment]
Have you considered using `middleware` for this? It doesn't require anything special on your host service, and if it's Vercel's, then it gets automatically deployed to the edge.

**Reply by kelko** [reply]
Good point, I haven’t thought about that. I’ll try it out for my use case.

**Reply by icyJoseph** [reply]
Yeah I also have encountered this kind of situation, but I think as more complex logic starts to be needed, it is perhaps best to use middleware and a protect the behaviour with a test.

**Reply by alexrequelme** [reply]
The problem is that the middleware is executed after getServerSideProps and getStaticProps, if you want to redirect a static page you have to use gSSP, if the nextconfig redirects() had a "has not" you wouldn't need that.
